Armadillo/src/main/java/com/scribd/armadillo/di/DownloadModule.kt

```diff
@@ -121,41 +121,50 @@ internal class DownloadModule {
     @Singleton
     @Provides
     @Named(Constants.DI.STANDARD_SECURE_STORAGE)
-    fun standardSecureStorage(context: Context): SharedPreferences {
-        val keys = MasterKeys.getOrCreate(
-            KeyGenParameterSpec.Builder("armadilloStandard", PURPOSE_ENCRYPT or PURPOSE_DECRYPT)
-                .setKeySize(256)
-                .setBlockModes(BLOCK_MODE_GCM)
-                .setEncryptionPaddings(ENCRYPTION_PADDING_NONE)
-                .build()
-        )
-        return EncryptedSharedPreferences.create(
-            "armadillo.standard.secure",
-            keys,
-            context,
-            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-        )
+    fun standardSecureStorage(context: Context): SharedPreferences? {
+        return try {
+            val keys = MasterKeys.getOrCreate(
+                KeyGenParameterSpec.Builder("armadilloStandard", PURPOSE_ENCRYPT or PURPOSE_DECRYPT)
+                    .setKeySize(256)
+                    .setBlockModes(BLOCK_MODE_GCM)
+                    .setEncryptionPaddings(ENCRYPTION_PADDING_NONE)
+                    .build()
+            )
+            EncryptedSharedPreferences.create(
+                "armadillo.standard.secure",
+                keys,
+                context,
+                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            )
+        } catch (ex: Exception) {
```

Comment:
What exception can actually happen here?

Reply:
Also I think this one should maybe fall back to unencrypted storage? It seems better than not being able to do anything.

Reply:
I dont know exactly, im following second hand reports on problems with EncryptedSharedPrefs. 

Failing into an insecure mode would not be appropriate.

Reply:
https://github.com/tink-crypto/tink/issues/535#issuecomment-912661574

Reply:
I'd recommend maybe doing something like this?

```
        val keySpec = KeyGenParameterSpec.Builder("armadilloStandard", PURPOSE_ENCRYPT or PURPOSE_DECRYPT)
            .setKeySize(256)
            .setBlockModes(BLOCK_MODE_GCM)
            .setEncryptionPaddings(ENCRYPTION_PADDING_NONE)
            .build()

        val keys = try {
            MasterKeys.getOrCreate(keyspec)
        } catch (ex: Exception){
                val keyStore = KeyStore.getInstance("AndroidKeyStore")
                keyStore.load(null)
                keyStore.deleteEntry("armadilloStandard")
                context.getSharedPreferences("armadillo.standard.secure", Context.MODE_PRIVATE).edit().clear().apply()
                MasterKeys.getOrCreate(keySpec)
        }

        return EncryptedSharedPreferences.create(
            "armadillo.standard.secure",
            keys,
            context,
            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
        )
```

Reply:
I think you may be fine to use `MasterKeys.AES256_GCM_SPEC` for both instead of a custom keyspec for each too, but your call